src/gluonnlp/data/transforms.py

```diff
@@ -693,3 +696,305 @@ def __call__(self, sample):
             Detokenized text
         """
         return self._processor.DecodePieces(sample)
+
+
+class BasicTokenizer():
+    r"""Runs basic tokenization
+
+    convert strings to unicode.
+    performs invalid character removal (e.g. control chars) and whitespace.
+    tokenize CJK chars. (https://en.wikipedia.org/wiki/CJK_Unified_Ideographs_(Unicode_block))
+    splits punctuation on a piece of text.
+    strips accents and convert to lower case.(If lower_case is true)
+
+    Parameters
+    ----------
+    lower_case : bool, default True
+        whether the text strips accents and convert to lower case.
+
+    Examples
+    --------
+    >>> tokenizer = gluonnlp.data.BasicTokenizer(lower_case=True)
+    >>> tokenizer(u" \tHeLLo!how  \n Are yoU?  ")
+    ['hello', '!', 'how', 'are', 'you', '?']
+    >>> tokenizer = gluonnlp.data.BasicTokenizer(lower_case=False)
+    >>> tokenizer(u" \tHeLLo!how  \n Are yoU?  ")
+    ['HeLLo', '!', 'how', 'Are', 'yoU', '?']
+
+    """
+
+    def __init__(self, lower_case=True):
+        self.lower_case = lower_case
+
+    def __call__(self, sample):
+        """
+
+        Parameters
+        ----------
+        sample: list(str)
+            The sentence to detokenize
+
+        Returns
+        -------
+        ret : str
+            Detokenized text
+        """
+        return self._tokenize(sample)
+
+    def _tokenize(self, text):
+        """Tokenizes a piece of text."""
+        text = _convert_to_unicode(text)
```

Comment:
Do we need to do encoding conversion inside the tokenizer? If the tokenizer only works for unicode text, why not simply require the user to call it with unicode text? Currently you assume that e.g. `text.decode('utf-8', 'ignore')` is appropriate for every user, but there are valid use cases where another error handler should be chosen, possibly even a custom one https://docs.python.org/3/library/stdtypes.html#bytes.decode